src/services/productService.ts

```diff
@@ -32,15 +33,15 @@ export const getProducts = async (query: any) => {
   };
 };
 
-export const getProductBySlug = (slug: string) => {
+export const getProductBySlug = (slug: string): Promise<Product | null> => {
   return productRepo.findProductBySlug(slug);
 };
 
 
 export const updateProductBySlug = async (
   slug: string,
   updates: UpdateProduct
-) => {
+): Promise<Product> => {
   const product = await productRepo.findProductBySlug(slug);
   if (!product) throw new Error('Product not found.');
```

Comment:
In a bid for consistency, you don't really need the full stop at the end

```suggestion
  if (!product) throw new Error('Product not found');
```